docs: Add how-to guide to create Intel Notebook (#163)
canonical/data-science-stack: docs/how-to/jupyter-notebook.rst

   tf.config.list_physical_devices('GPU')

Create an Intel GPU-enabled notebook
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> You can create an Intel GPU-enabled Jupyter Notebook with IPEX or ITEX.

Reviewer: Can we add a link under "IPEX" and "ITEX" for referencing the user to further info?

Author: posted the question on the MM channel, not sure which links we can use here

.. note::

>    To launch an Intel GPU-enabled notebook, you must first :doc:`Enable Intel GPUs <../how-to/enable-gpus/enable-intel-gpu>`.

Reviewer: Let's use `:ref:` + the tag here for the internal linking. I committed this change but probably you didn't refresh your local PR with my change and got overwritten.

Author: done

To see the list of available Intel images, run:

.. code-block:: bash

   dss create --help | grep intel

You should see an output similar to this:

.. code-block:: bash

        - intel-pytorch = intel/intel-extension-for-pytorch:2.1.20-xpu-idp-jupyter
        - intel-tensorflow = intel/intel-extension-for-tensorflow:2.15.0-xpu-idp-jupyter

Select one of them and create a notebook as follows:

.. code-block:: bash

   dss create my-itex-notebook --image=intel-tensorflow

> Confirm the devices are detected and usable by running:

Reviewer: ```suggestion
Confirm the GPU is detected and usable by running:
```
Let's focus on the GPU, not devices in general

Author: done

.. code-block:: python

   import tensorflow as tf

   tf.config.experimental.list_physical_devices()

> Example of the expected output for a host system containing an Intel CPU and a single Intel GPU:

Reviewer: ```suggestion
For example, you should expect an output like the following for a host system containing an Intel CPU and a single Intel GPU:
```

Author: done

.. code-block:: python

    [PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU'), PhysicalDevice(name='/physical_device:XPU:0', device_type='XPU')]

.. note::

>     Intel calls GPU+CPU mode XPU

Reviewer: ```suggestion
    Note that Intel denotes XPU the combination of an Intel CPU with GPU.
```

Author: done
